### DISCUSSION TITLE
How to provide dynamic configuration for Camel Pods

### DISCUSSION THREAD

**Discussion by aleksanderson** [apache/camel-karavan#795] [Q&A] [answered]
Hello,

First of all, awesome work that you're doing with Kamel. I'm currently playing with the tool and have a question

What should be the method of providing a dynamic configuration for camel services running in the pod using Karavan? Is there a way to bind them to the K8s' ConfigMap/Secret services like it is done for Camel K - https://camel.netlify.app/camel-k/latest/configuration/configmap-secret.html

Thank you

**Comment by davsclaus** [comment]
Karavan lets you use SB or Quarkus as the runtime, and use them 100% as-is. This means you can configure k8s "stuff" in their standard way.

You can then just have an `application.properties` file, where you can configure it
https://quarkus.io/guides/kubernetes-config

However I am sure we can make something better in Karavan / camel-jbang. So if you are able to provide a small example that is much better to discuss and see what can be done.

**Comment by mgubaidullin** [comment, marked as answer]
![karavan-secrets](https://github.com/apache/camel-karavan/assets/1379213/d95d35d8-d66e-4297-8844-d3613209bdab)

**Comment by patrick-krivrog** [comment]
@mgubaidullin 
Hi there!
Help me please, I don't understand why I see my secrets from k8s in drop-downs but not in the configuration tab? How I can customise displaying Statuses/Secrets/ConfigMaps on the configuration tab?
<img width="1474" alt="image" src="https://github.com/apache/camel-karavan/assets/126600868/b11845ef-fe18-471d-9a18-dbc40de88c72">
<img width="1470" alt="image" src="https://github.com/apache/camel-karavan/assets/126600868/75050803-6a57-402e-879e-77314e47a1d2">

**Reply by mgubaidullin** [reply]
This part is not yet implemented )

**Reply by patrick-krivrog** [reply]
are looking forward to...

**Reply by mgubaidullin** [reply]
still debating... ConfigMaps and Secrets provided by Kubernetes Dashboard and we do not want to reinvent it in Karavan.
What is your use-case for it?

**Reply by patrick-krivrog** [reply]
Provide the Karavan web app as an ecosystem for a Proxy Server of hundreds of routes for AFCS type system. So, it is nice to see the list of secrets and configmaps at least in read-only mode. Or the possibility of hiding the specific tabs in settings or config file.

**Reply by mgubaidullin** [reply]
AFCS is an automatic flight control system?

**Reply by patrick-krivrog** [reply]
Automated Fare Collection System
